rfcs/2021-01-05-typography-scaling.md

```diff
@@ -0,0 +1,223 @@
+
+### **Current Limitations**
+There are two major limitations while scaling typography to multiple breakpoints,
+1. **It does not cater to landing/marketing pages.** Which are generally much larger compared to the size-ratio on the dashboard (Maybe a [Major Third](https://type-scale.com/?size=14&scale=1.125&text=A%20Visual%20Type%20Scale&font=Lato&fontweight=400&bodyfont=Poppins&bodyfontweight=400&lineheight=1.65&backgroundcolor=white&fontcolor=%23333&preview=false) scale will work better). This is currently being looked at by the comm. design team based on the same principles.
+2. **Same type scaling across all devices/breakpoints.** After the type scale is defined, we also need to define the scale across multiple device sizes. There are few approaches that we've came up with and after research from other DS/products like *GitHub*, *Eightshapes*, *Paystack*, *Stripe*, and a few more we've observed that for dashboards generally tweaking the **heading** sizes works best and the base font-size will remain same for all other devices/breakpoints.
+
+We'll look into the responsive type scale and see what approach fits best for our use case.
+
+>📝 Note: **Responsive Typography** will scale in segments. Which means if `h1` at `1440px` is `40px`, then at `991px` it will become `32px` directly rather than going from `40px -> 38px -> 36px -> 34px -> 32px...`.
+
+### **Our Approach**
+>**Multiple type-scaling across breakpoints**
+
+Our product is majorly used on mobiles/tablets/desktops. Therefore, We'll  consider only two breakpoints for the typography scale, one for mobiles & one for both desktops & tablets. Which is for devices having width `>=991px` (approx.)
+
+![Scaling: Multiple type scales ratio](./images/typography/typography-multi-type-scaling.png)
+
+In this type of scaling the modular scale changes on every breakpoint. For example: **mobiles** will use `Minor Third (1.2)` and **desktops/tablets** will use `Custom Scale (1.1)`.
+
+>📝 Note: Here we're rounding-off all the numbers for the sake of simplicity and to align the content according to the grid-system. Also, various browsers treat fractional font sizes differently.
+
+### **Example:**
+This is how it will look like across desktop & mobile.
+
+- **Type Unit Scale**
+![Type Unit Scale (Typography)](./images/typography/type-unit-scale.png)
+
+- **Typography Scale**
+![Typography Scale on Desktop vs Mobile (Typography)](./images/typography/typography-scale.png)
+
+- **Another visual indicator**
+![Scaling: Desktop & Mobile](./images/typography/type-scale-blade.png)
+
+
+# Drawbacks/Constraints
+Few things to consider here:
+- As we are moving towards adaptive-layout so we need to consider - should we also support responsive layout? For both cases, I don't see any challenges as the typography system remains the same.
+- How will this be managed on the tech-side? This might create more work as we'll have to maintain more scales,
+  - Landing/Marketing page (mobile & desktop/tablets)
```

Comment:
not sure if we also now need another scale for marketing

Reply:
Maybe we can rephrase it:
If any product wants to bring in their own typography scale then they need to override the current typography scale provided by blade. This also means that you cannot add/remove levels from the base scale